Modal refactor

theyapps opened the pull request:
Refactored modal to increase re-usability.

theyapps commented on spa/src/components/modals/services/ModalService.vue:
```diff
@@ -0,0 +1,15 @@
+  methods: {
+    open(component, props = {}) {
+      return new Promise((resolve, reject) => {
+        console.log("ModalService open");
```
Remove console log

theyapps commented on spa/src/components/modals/TestModal.vue:
```diff
@@ -0,0 +1,23 @@
+<template>
```
Delete testmodal

TheMikeSanto commented on spa/src/components/modals/Modal.vue:
```diff
@@ -0,0 +1,51 @@
+export default Vue.extend({
+  name: "Modal",
+  components: {},
+  data: () => {
```
Could shorten this up:
`data: () => ({}),`

theyapps replied:
done

TheMikeSanto commented on spa/src/components/modals/mixins/ModalMixin.ts:
```diff
@@ -0,0 +1,12 @@
+export default {
```
This is awesome!

TheMikeSanto commented on spa/src/components/modals/services/ModalService.vue:
```diff
@@ -0,0 +1,15 @@
+  methods: {
+    open(component, props = {}) {
+      return new Promise((resolve, reject) => {
+        console.log("ModalService open");
```
Get rid of this log statement

theyapps replied:
Done!

TheMikeSanto reviewed (changes requested):
This looks like what we need! A few minor changes to address, and then it's ready to merge. Thank you!!

theyapps commented on spa/src/components/modals/Modal.vue:
```diff
@@ -0,0 +1,51 @@
+        bg-black
+        overflow-y-hidden" style="width:1024px;">
+      <div class="flex
+      flex-row-reverse
```
done

TheMikeSanto commented:
@theyapps I believe this is ready to merge, but I haven't had time to pull the PR and actually boot it up and test. I will get to that this week, I promise. :)

theyapps commented:
> @theyapps I believe this is ready to merge, but I haven't had time to pull the PR and actually boot it up and test. I will get to that this week, I promise. :)

Sounds good.

dburleson commented:
@theyapps has discussed, the code functions well. But, there are a number of coding standards issues, mainly incorrect linebreak style and some semicolons missing.

Once these are resolved, I'll happily get this merged.

theyapps commented:
> @theyapps has discussed, the code functions well. But, there are a number of coding standards issues, mainly incorrect linebreak style and some semicolons missing.
> 
> Once these are resolved, I'll happily get this merged.

I have updated the line endings and added semicolons
